auth/auth.go

```diff
@@ -265,22 +330,30 @@ func (auth *Authenticator) UpdateSequenceNumber(p Principal, seq uint64) error {
 }
 
 // Invalidates the channel list of a user/role by saving its Channels() property as nil.
-func (auth *Authenticator) InvalidateChannels(p Principal) error {
+func (auth *Authenticator) InvalidateChannels(p Principal, invalSeq uint64) error {
 	invalidateChannelsCallback := func(p Principal) (updatedPrincipal Principal, err error) {
 
 		if p == nil || p.Channels() == nil {
 			return p, base.ErrUpdateCancel
 		}
 
 		base.Infof(base.KeyAccess, "Invalidate access of %q", base.UD(p.Name()))
+
+		if p.PreviousChannels() == nil {
```

Comment:
I'll put in a comment too but for first question:
- Previous channels is only set once because we only need to know the information since the user last pulled data. If there are a bunch of grants removed and added again between pulls we don't really care about those. 

Second:
- Yeah I've had a think and I can't think of a scenario where you'd end up with both being non-nil but as you say its defensive so doesn't hurt.

Reply:
Removed Previous